ResearchKit/Common/ORKTaskViewController.m

```diff
@@ -1245,6 +1304,51 @@ - (ORKStep *)stepAfterStep:(ORKStep *)step {
     return [self.task stepAfterStep:step withResult:[self result]];
 }
 
+#pragma mark - ORKReviewStepViewControllerDelegate
+
+- (nonnull NSArray *)reviewStepViewController:(ORKReviewStepViewController *)reviewStepViewController
+                           stepsForReviewStep:(ORKReviewStep *)reviewStep {
+    NSMutableArray *steps = [[NSMutableArray alloc] init];
+    if (reviewStep.isStandalone) {
+        for (ORKStep *step in reviewStep.steps) {
+            if ([reviewStepViewController.reviewDelegate respondsToSelector:@selector(reviewStepViewController:shouldReviewStep:)] && [reviewStepViewController.reviewDelegate reviewStepViewController:reviewStepViewController shouldReviewStep:step]) {
+                [steps addObject:step];
+            }
+        }
+    } else {
+        ORKStep *nextStep = reviewStep;
+        do {
+            switch (reviewStep.reviewDirection) {
+                case ORKReviewStepReviewDirectionForward:
+                    nextStep = [self stepAfterStep:nextStep];
+                    break;
+                case ORKReviewStepReviewDirectionReverse:
+                    nextStep = [self stepBeforeStep:nextStep];
+                    break;
+            }
+            if (nextStep && ![nextStep isKindOfClass:[ORKReviewStep class]]) {
+                BOOL shouldAddStep = [reviewStepViewController.reviewDelegate respondsToSelector:@selector(reviewStepViewController:shouldReviewStep:)] && [reviewStepViewController.reviewDelegate reviewStepViewController:reviewStepViewController shouldReviewStep:nextStep];
+                if (shouldAddStep) {
+                    [steps addObject:nextStep];
+                }
+            }
+        } while (nextStep || [nextStep isKindOfClass:[ORKReviewStep class]]);
+    }
+    return (reviewStep.reviewDirection == ORKReviewStepReviewDirectionReverse) ? [[steps reverseObjectEnumerator] allObjects] : [steps copy];
+}
+
+- (BOOL)reviewStepViewController:(ORKReviewStepViewController *)reviewStepViewController
+                shouldReviewStep:(ORKStep *)step {
+    return ![step isKindOfClass:[ORKReviewStep class]] || ![step isKindOfClass:[ORKActiveStep class]];
```

Comment:
1. only questionStep and formStep are allowed
2. should expose this `shouldReviewStep:` to taskVC delegate to opt out any step the developer don't want to include.